crates/supervisor/core/src/supervisor.rs

```diff
@@ -210,8 +212,17 @@ impl SupervisorService for Supervisor {
         self.config.dependency_set.dependencies.keys().copied()
     }
 
-    fn super_head(&self, _chain: ChainId) -> Result<SuperHead, SupervisorError> {
-        todo!("implement call to ChainDbFactory")
+    fn super_head(&self, chain: ChainId) -> Result<SuperHead, SupervisorError> {
+        let db = self.database_factory.get_db(chain)?;
+
+        let l1_source = db.get_current_l1()?;
+        let local_unsafe = db.get_safety_head_ref(SafetyLevel::Unsafe)?;
+        let cross_unsafe = db.get_safety_head_ref(SafetyLevel::CrossUnsafe)?;
```

Comment:
It would be better to create a `get_super_head` method on the `chainDB` and use that here. `HeadRefStorageReader::get_super_head` can use a single `tx` to get the data from the store thereby making sure we have the consistent view of the data.

Reply:
I thought so too, however we can do this as a follow up I thought in MVP 3, unless it is a non-spec'd assumption of the Go implementation which hence will lead to e2e tests using this rpc method failing

Reply:
Yep, I a happy with this being the first iteration.